internal/kafka/internal/services/kafka.go

```diff
@@ -287,12 +287,19 @@ func (k *kafkaService) AssignInstanceType(owner string, organisationId string) (
 		return "", errors.NewWithCause(errors.ErrorGeneral, factoryErr, "unable to check quota")
 	}
 
-	hasRhosakQuota, err := quotaService.CheckIfQuotaIsDefinedForInstanceType(owner, organisationId, types.STANDARD)
-	if err != nil {
-		return "", err
-	}
-	if hasRhosakQuota {
-		return types.STANDARD, nil
+	for _, instanceType := range k.kafkaConfig.SupportedInstanceTypes.Configuration.SupportedKafkaInstanceTypes {
+		if instanceType.Id == types.DEVELOPER.String() {
+			continue
+		}
+		for _, bm := range instanceType.SupportedBillingModels {
+			hasRhosakQuota, err := quotaService.CheckIfQuotaIsDefinedForInstanceType(owner, organisationId, types.KafkaInstanceType(instanceType.Id), bm)
+			if err != nil {
+				return "", err
+			}
+			if hasRhosakQuota {
+				return types.KafkaInstanceType(instanceType.Id), nil
+			}
+		}
 	}
 
 	return types.DEVELOPER, nil
```

Comment:
A side comment about this which does not seem strictly related to this PR: It returns `developer` but if `developer` is not defined in the supported kafka instances configuration then this would be incorrect.

Reply:
Agree. Currently, the `developer` instance type is an alien beast: if you don't have any quota configured, you can create a `developer`. No configuration needs to be present, and no quota either.